src/BlockTypes/ProductGalleryPager.php

```diff
@@ -37,96 +39,128 @@ protected function get_block_type_uses_context() {
+			$html = $this->render_pager( $product_gallery_images_ids, $pager_display_mode, $number_of_thumbnails );
+
+			return sprintf(
+				'<div %1$s>
+					%2$s
+				</div>',
+				$wrapper_attributes,
+				$html
+			);
 		}
 	}
 
 	/**
-	 * Renders the digits pager HTML.
+	 * Renders the pager for the product gallery.
 	 *
-	 * @return string The rendered digits pager HTML.
+	 * @param  array  $product_gallery_images_ids An array of image IDs for the product gallery.
+	 * @param  string $pager_display_mode         The display mode for the pager.
+	 * @param  int    $number_of_thumbnails       The number of thumbnails to display in the pager.
+	 * @return string|null The rendered pager HTML, or null if the pager should not be displayed.
 	 */
-	private function render_digits_pager() {
-		return sprintf(
-			'<ul class="wp-block-woocommerce-product-gallery-pager__pager">
-				<li class="wp-block-woocommerce-product-gallery__pager-item is-active">1</li>
-				<li class="wp-block-woocommerce-product-gallery__pager-item">2</li>
-				<li class="wp-block-woocommerce-product-gallery__pager-item">3</li>
-				<li class="wp-block-woocommerce-product-gallery__pager-item">4</li>
-			</ul>'
-		);
+	private function render_pager( $product_gallery_images_ids, $pager_display_mode, $number_of_thumbnails ) {
+		if ( $number_of_thumbnails < 2 || 'off' === $pager_display_mode ) {
+			return null;
+		}
+
+		return $this->render_pager_pages( $product_gallery_images_ids, $number_of_thumbnails, $pager_display_mode );
 	}
 
 	/**
-	 * Renders the dots pager HTML.
+	 * Renders the pager pages for the product gallery.
 	 *
-	 * @return string The rendered dots pager HTML.
+	 * @param  array  $product_gallery_images_ids An array of image IDs for the product gallery.
+	 * @param  int    $number_of_thumbnails The number of thumbnails to display in the pager.
+	 * @param  string $pager_display_mode The display mode for the pager. Defaults to 'dots'.
+	 * @return string The rendered pager pages HTML.
 	 */
-	private function render_dots_pager() {
+	private function render_pager_pages( $product_gallery_images_ids, $number_of_thumbnails, $pager_display_mode = 'dots' ) {
+		$html = '';
+
+		foreach ( $product_gallery_images_ids as $key => $product_gallery_image_id ) {
+			if ( $key >= $number_of_thumbnails ) {
+				break;
+			}
+
+			$is_first_pager_item = 0 === $key;
+			$pager_item          = sprintf(
+				'<li class="wp-block-woocommerce-product-gallery__pager-item %2$s">%1$s</li>',
+				'dots' === $pager_display_mode ? $this->get_dot_icon( $is_first_pager_item ) : $key + 1,
+				$is_first_pager_item ? 'is-active' : ''
+			);
+			$p                   = new \WP_HTML_Tag_Processor( $pager_item );
+
+			if ( $p->next_tag() ) {
+				$p->set_attribute(
+					'data-wc-context',
+					wp_json_encode(
+						array(
+							'isSelected'  => $is_first_pager_item,
+							'woocommerce' => array( 'imageId' => strval( $product_gallery_image_id ) ),
+						)
+					)
```

Comment:
Everything should be inside the woocommerce key. Also, is it necessary `strval` since we converting the ID in the `get_product_gallery_image_ids` function